apps/src/blockly/customBlocks/googleBlockly/behaviorBlocks.js

```diff
@@ -0,0 +1,256 @@
+      'procedure_caller_update_shape_mixin',
+      'procedure_caller_context_menu_mixin',
+      'procedure_caller_onchange_mixin',
+      'procedure_callernoreturn_get_def_block_mixin',
+    ],
+    mutator: 'behavior_get_mutator',
+  },
+  {
+    type: 'sprite_parameter_get',
+    message0: '%1',
+    args0: [
+      {
+        type: 'field_label',
+        name: 'VAR',
+        variable: '%{BKY_VARIABLES_DEFAULT_NAME}',
+        text: msg.thisSprite(),
+      },
+    ],
+    output: 'Sprite',
+    style: 'sprite_blocks',
+    helpUrl: '%{BKY_VARIABLES_GET_HELPURL}',
+    tooltip: '%{BKY_VARIABLES_GET_TOOLTIP}',
+    extensions: ['contextMenu_variableSetterGetter'],
+  },
+]);
+
+GoogleBlockly.Extensions.registerMutator(
+  'behavior_def_mutator',
+  behaviorDefMutator
+);
+
+// This extension adds an SVG frame around behavior definition blocks.
+// Not used when the modal function is enabled.
+GoogleBlockly.Extensions.register('behaviors_block_frame', function () {
+  if (!useModalFunctionEditor && !this.workspace.noFunctionBlockFrame) {
+    this.functionalSvg_ = new BlockSvgFrame(
+      this,
+      msg.behaviorEditorHeader(),
+      'blocklyFunctionalFrame'
+    );
+
+    this.setOnChange(function () {
+      if (!this.isInFlyout) {
+        this.functionalSvg_.render(this.svgGroup_, this.RTL);
+      }
+    });
+  }
+});
+
+GoogleBlockly.Extensions.registerMutator(
+  'behavior_get_mutator',
+  behaviorGetMutator
+);
+
+/**
+ * Constructs the blocks required by the flyout for the procedure category.
+ * Modeled after core Blockly procedures flyout category, but excludes unwanted blocks.
+ * Derived from core Google Blockly:
+ * https://github.com/google/blockly/blob/5a23c84e6ef9c0b2bbd503ad9f58fa86db1232a8/core/procedures.ts#L202-L287
+ * @param {WorkspaceSvg} workspace The workspace containing procedures.
+ * @returns an array of XML block elements
+ */
+export function flyoutCategory(workspace) {
+  const blockList = [];
+
+  const newBehaviorButton = {
+    kind: 'button',
+    text: msg.createBlocklyBehavior(),
+    callbackKey: 'createNewBehavior',
+  };
+
+  const behaviorDefinitionBlock = {
+    kind: 'block',
+    type: 'behavior_definition',
+    fields: {
+      NAME: Blockly.Msg.PROCEDURES_DEFNORETURN_PROCEDURE,
+    },
+  };
+
+  const createNewBehavior = function () {
+    // Everything here is place-holder code that should be replaced with a
+    // call to open the behavior editor with a new defintion block.
+    // Until then, we just create a block under all existing blocks on the
+    // main workspace.
+
+    createAndCenterNewDefBlock(behaviorDefinitionBlock);
+  };
+  if (useModalFunctionEditor) {
+    workspace.registerButtonCallback('createNewBehavior', createNewBehavior);
+    blockList.push(newBehaviorButton);
+  } else {
+    blockList.push(behaviorDefinitionBlock);
+  }
+  blockList.push(...getCustomCategoryBlocksForFlyout('Behavior'));
+
+  const allWorkspaces = Blockly.Workspace.getAll().filter(
```

Comment:
I was assuming so. My thought process:
1. This code is populating blocks in the behaviors flyout category.
2. To do so, it gets a list of all behavior names from any top-level blocks on any (non-flyout) workspace with the `behavior_definition` type. 
3. Behavior block definitions can exist:
- On the main workspace (if the function editor is disabled and the user has custom functions defined?)
- On the hidden workspace (if the function editor is enabled and/or we're talking about out-of-the-box behaviors, whose definitions will always be added to the hidden workspace?)
- On the editor workspace (?)

Since this would also cover the editor workspace, I wondered for a second if we could run into issues with duplicate behavior definitions existing on both the editor workspace and the hidden workspace causing two flyout blocks for a single behavior, but your modal editor PR only uses blocks of type `procedures_defnoreturn`, so not yet at least?

Reply:
If something exists on the editor workspace, it should also exist on the hidden workspace, since the editor workspace is ephemeral. Once my change is in we will have a reference to both and could just search those two workspaces instead of all of them? But this is probably fine for now.